src/App.tsx

```diff
@@ -15,48 +24,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const getSortedGoods = (condition: Sort, isReversed: boolean) => {
+  const goodsCopy = [...goodsFromServer];
+
+  goodsCopy.sort((a, b) => {
+    switch (condition) {
+      case Sort.ALPHABETICALLY:
+        return a.localeCompare(b);
+
+      case Sort.BY_LENGTH:
+        return a.length - b.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? goodsCopy.reverse() : goodsCopy;
+};
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(Sort.DEFAULT);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const goods: Good[] = getSortedGoods(sortField, isReversed);
+
+  const reset = () => {
```

Comment:
You should choose a better name for this handler:
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364